plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseSharedMetastoreTest.java

```diff
@@ -128,4 +134,57 @@ public void testShowSchemas()
                 showCreateIcebergWithRedirectionsSchema,
                 getExpectedIcebergCreateSchema("iceberg_with_redirections"));
     }
+
+    @Test
+    public void testTimeTravelAsOfVersionWithRedirection()
```

Comment:
`AsOfVersion` is a bit misleading because the test verifies also the functionality of `FOR TIMESTAMP ...` syntax.

Reply:
right initially I wanted to have 2 different tests then I changed my mind